File: ruby/word-count/word_count.rb
class Phrase
  VERSION = 1
  attr_reader :word_count

  def initialize(input)
    @word_count = input.split(/\W+/).reduce({ }) do |memo, word|
      word = word.downcase
      memo[word] ||= 0
      memo[word] += 1
      memo
    end
  end
end

File: java/word-count/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;

public class WordCount {
  public Map<String, Integer> Phrase(String input) {
    Map<String, Integer> res = new HashMap<String, Integer>();

    for (String word : input.toLowerCase().split("\\W+")) {
      incrementCount(res, word);
    }

    return res;
  }

  private void incrementCount(Map<String, Integer> map, String word) {
    if (!map.containsKey(word)) map.put(word, 0);

    int crnt = map.get(word);

    map.put(word, crnt + 1);
  }
}
